pkg/sdk/go/sdk.go

```diff
@@ -349,3 +367,33 @@ func (sdk mfSDK) sendThingRequest(req *http.Request, key, contentType string) (*
 
 	return sdk.client.Do(req)
 }
+
+func (sdk mfSDK) parseUserFilteredValues(baseURL, endpoint string, filter UserPageMetadata) (string, error) {
```

Comment:
This can be improved by Marshaling `UserPageMetadata` in a `map[string]interface` and then adding query parameters to the URL using a for loop and a switch case. Something like that: 
```
        var queryParams := ""
	for key, val := range pageMeta {
		switch k {
		case
			"offset",
			"limit",
			"name",
			"email":
			queryParams = fmt.Sprintf(`%s&%s=%s `, queryParams, key, val)
		case "metadata":
                       ...
		}
	}
	return fmt.Sprintf("%s/%s?%s, baseURL, endpoint, queryParams)
```

You have to also check if it's or not the first query parameter to handle properly `?` and `&`.